### ISSUE TITLE
Not seeing errors from CLI with custom ruleset

### ISSUE THREAD

**Issue by thelostspore** [squizlabs/PHP_CodeSniffer#1108]
Following the example in the phpcs [wiki page on custom rulesets](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml), I created a [custom ruleset](https://github.com/thelostspore/phpcs-ruleset-help/blob/master/phpcs.xml) and am attempting to run a phpcs command like this:

```
phpcs -snp --standard=./phpcs.xml --report=summary example.php
```

However, attempting to validate a [php file with obvious errors](https://github.com/thelostspore/phpcs-ruleset-help/blob/master/example.php), I cannot get phpcs to emit any errors. 

I'm interested in using a custom ruleset to set the `type` and `severity` for specific sniffs. Any help would be very much appreciated. I've created a [repo]() for the examples I've used above.

**Comment by gsherwood** [comment]
When you set severity and type in a ruleset, you need to set it on a message. In your ruleset, you are setting it on an entire sniff and not a specific message.

For example, this entry:
```xml
<rule ref="Squiz.Strings.DoubleQuoteUsage">
    <type>error</type>
    <severity>10</severity>
 </rule>
```
Should be this:
```xml
<rule ref="Squiz.Strings.DoubleQuoteUsage.NotRequired">
    <type>error</type>
    <severity>10</severity>
 </rule>
```
You can see the error messages codes by using the `-s` command line argument.

Note that in the 3.0 version (currently released as 3.0.0a1) you can set the type and severity of an entire sniff. When running you ruleset over that test file using the 3.0.0a1 version, I get this output:
```
FILE: temp.php
------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 8 ERRORS AFFECTING 5 LINES
------------------------------------------------------------------------------------------------------------------------------------------------------
 3 | ERROR | [ ] Line exceeds maximum limit of 30 characters; contains 34 characters (Generic.Files.LineLength.MaxExceeded)
 4 | ERROR | [ ] Line exceeds 20 characters; contains 30 characters (Generic.Files.LineLength.TooLong)
 4 | ERROR | [x] String "double quoted string!!!" does not require double quotes; use single quotes instead
   |       |     (Squiz.Strings.DoubleQuoteUsage.NotRequired)
 5 | ERROR | [ ] Line exceeds maximum limit of 30 characters; contains 111 characters (Generic.Files.LineLength.MaxExceeded)
 5 | ERROR | [x] String "this line is long and should fail! this line is long and should fail! this line is long and should fail!" does not require
   |       |     double quotes; use single quotes instead (Squiz.Strings.DoubleQuoteUsage.NotRequired)
 7 | ERROR | [ ] Line exceeds 20 characters; contains 22 characters (Generic.Files.LineLength.TooLong)
 8 | ERROR | [ ] Duplicate class name "DuplicateClass" found; first defined in /Users/gsherwood/Sites/Projects/PHP_CodeSniffer/temp.php on line 7
   |       |     (Generic.Classes.DuplicateClassName.Found)
 8 | ERROR | [ ] Line exceeds 20 characters; contains 22 characters (Generic.Files.LineLength.TooLong)
------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 24ms; Memory: 4Mb
```
But that's only in the 3.0 version, so you'll need to set the type and severity of individual messages if you are using the 2.x version.

Does that all make sense?

**Comment by thelostspore** [comment]
Thanks @gsherwood, makes sense. Appreciate the example too.

In general, is there documentation for sniff messages? For example, in the case of the [DoubleQuotedString](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Squiz/Sniffs/Strings/DoubleQuoteUsageSniff.php) sniff, how would you know what messages are available?  

Also, is there a target date for the stable release of 3.x?

**Comment by aik099** [comment]
No documentation for sniff messages. I think best approach would be to try running with all messages and see which ones you don't like and disable then on the fly.

**Comment by gsherwood** [comment]
> Also, is there a target date for the stable release of 3.x?

Nope. Ready when it's ready, heavily based on getting people to actually test it.

Glad the explanation made sense though.
